tools/hardhat/submit-application.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# Default network value
+DOCKER_DIR="../../docker"
+NATIVE_DIR="../../native"
+HOME_DIR="$(pwd)"
+NETWORK="testnet" # default
+FLOW=""
+
+# install dependencies
+npm i
+
+# Function to display usage
+usage() {
+  echo "Usage: $0 [--network <network>] --flow <flow_value>"
+  exit 1
+}
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --network)
+      NETWORK="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --flow)
+      FLOW="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+# Check if mandatory field is set
+if [ -z "$FLOW" ]; then
+  echo "Error: --flow is a mandatory flag."
+  usage
+fi
+
+# Output the values used
+echo "Network: $NETWORK"
+echo "Flow: $FLOW"
+
+# Specify the path to the *.env file (e.g. testnet.env)
+env_file="$NETWORK.env"
+
+if [ "$FLOW" = "docker" ]; then
+    cd $DOCKER_DIR || { echo "Failed to navigate to docker directory"; exit 1; }
+    output=$(./v-kit.sh validator-info)
+    # Extract MEZOD_MONIKER from the env file
+    moniker=$(grep "^MEZOD_MONIKER=" "$env_file" | awk -F'=' '{print $2}')
+elif [ "$FLOW" = "native" ]; then
+    cd $NATIVE_DIR || { echo "Failed to navigate to native directory"; exit 1; }
+    output=$(sudo ./v-kit.sh --validator-info)
+    # Extract MEZOD_MONIKER from the env file
+    moniker=$(grep "^MEZOD_MONIKER=" "$env_file" | awk -F'=' '{print $2}')
+else
+    echo "Error: Invalid flow value. Please use 'docker' or 'native'."
+    exit 1
+fi
+cd $HOME_DIR
+
+# Extract "Validator address"
+signer=$(echo "$output" | grep "Validator address" | awk -F': ' '{print $2}')
+
+# Extract "Validator consensus address"
+conspubkey=$(echo "$output" | grep "Validator consensus address" | awk -F': ' '{print $2}')
```

Comment:
Looks like `validator-info` from the kit wrongly names that field. This should be `Validator consensus pubkey`. Same for `Validator network address` that is taken from `priv_validator_key.json`. The network (p2p) address is derived from `node_key.json`. The address derived from `priv_validator_key.json` is the consensus address. I totally missed it during the review of the kits. cc @thevops @tscrond

Reply:
I'll provide a PR for docker deployment

Reply:
@dimpar @lukasz-zimnoch https://github.com/mezo-org/mezod/pull/344

Reply:
This was changed in the entrypoint https://github.com/mezo-org/mezod/pull/344. However, let's stick with old values for now as this change in the entrypoint.sh will be included in the next release of the Docker image. We can leave a TODO to change this in the `submit-application.sh` then.

Reply:
Added TODO [9c9aa83](https://github.com/mezo-org/validator-kit/pull/5/commits/9c9aa83c9e5cee9e9ed848ab242c306e9ea39dde)